src/centralized_telescope.rs

```diff
@@ -0,0 +1,419 @@
+//! ALBA's bounded DFS scheme using Blake2b as hash function.
+//! (c.f. Section 3.2.2 of Alba paper)
+
+use crate::utils;
+use blake2::{Blake2s256, Digest};
+use std::{f32::consts::LOG2_E, f64::consts::E};
+
+const DATA_LENGTH: usize = 32;
+const DIGEST_SIZE: usize = 32;
+
+type Element = [u8; DATA_LENGTH];
+type Hash = [u8; DIGEST_SIZE];
+
+/// Setup input parameters
+#[derive(Debug)]
+pub struct Params {
+    /// Soundness security parameter
+    pub lambda_sec: f64,
+    /// Completeness security parameter
+    pub lambda_rel: f64,
+    /// Approximate size of set Sp to lower bound
+    pub n_p: u64,
+    /// Target lower bound
+    pub n_f: u64,
+}
+
+/// Setup output parameters
+#[derive(Debug, Clone)]
+pub struct Setup {
+    /// Security parameter
+    pub sec_param: u64,
+    /// Approximate size of set Sp to lower bound
+    pub n_p: u64,
+    /// Proof size (in Sp elements)
+    pub u: u64,
+    /// Proof max counter
+    pub r: u64,
+    /// Proof max 2nd counter
+    pub d: u64,
+    /// Probability q
+    pub q: f64,
+    /// Computation bound
+    pub b: u64,
+}
+impl Setup {
```

Comment:
This version of Setup seems much better and more user-friendly.
However, I would prefer to see the `new` function at the bottom.

Reply:
I am not sure to understand what you mean? The `new` function is at the bottom (of the impl Setup), do you mean you want the implementation of Setup to be at the bottom (first we define all structures then the implementations)?
This should be covered in https://github.com/cardano-scaling/alba/issues/46, where we will split the file depending on the structure.

Reply:
I think Gamze means that the `new()` function should be defined first in the `impl Setup` block

Reply:
nope. Not at the bottom, but at the top, I would say. And, should be addressed in #46 for sure